astrokat/test/test_offline_observe.py

```diff
@@ -1,111 +1,93 @@
+from __future__ import absolute_import
 from __future__ import print_function
+
+
 import unittest2 as unittest
-import sys
+from mock import patch
 
-from astrokat import observe_main, simulate
 
-import logging
-if sys.version_info[0] == 3:
-    from io import StringIO
-else:
-    from StringIO import StringIO
+from .testutils import LoggedTelescope, execute_observe_main
 
+from six.moves import StringIO
 
-class test_astrokat_yaml(unittest.TestCase):
 
-    def setUp(self):
-        user_logger = logging.getLogger('astrokat.simulate')
+@patch('astrokat.observe_main.Telescope', LoggedTelescope)
+class TestAstrokatYAML(unittest.TestCase):
 
-        # remove current handlers
-        for handler in user_logger.handlers:
-            user_logger.removeHandler(handler)
+    def setUp(self):
+        LoggedTelescope.user_logger_stream = StringIO()
 
-        self.string_stream = StringIO()
-        out_hdlr = logging.StreamHandler(self.string_stream)
-        formatter = logging.Formatter('%(asctime)s - %(message)s')
-        formatter.formatTime = simulate.sim_time
-        out_hdlr.setFormatter(formatter)
-        out_hdlr.setLevel(logging.TRACE)
-        user_logger.addHandler(out_hdlr)
-        user_logger.setLevel(logging.INFO)
 
     def test_targets_sim(self):
-        observe_main.main(['--yaml',
-                           'astrokat/test/test_obs/targets-sim.yaml'])
+        execute_observe_main('test_obs/targets-sim.yaml')
 
         # get result and make sure everything ran properly
-        result = self.string_stream.getvalue()
-
-        # TODO: restore this check after working out an appropriate start-time
-        # in UTC with Ruby
-
+        result = LoggedTelescope.user_logger_stream.getvalue()
         self.assertIn('Single run through observation target list', result, 'Single run')
-        # self.assertIn('target0_radec observed for 10.0 sec',
-        #               result, 'target0_radec observed for 10.0 sec')
-        # self.assertIn('target1_azel observed for 10.0 sec',
-        #               result, 'target1_azel observed for 10.0 sec')
-        # self.assertIn('target2_gal observed for 10.0 sec',
-        #               result, 'target2_gal observed for 10.0 sec')
+        self.assertIn('target0_radec observed for 10.0 sec',
+                      result, 'target0_radec observed for 10.0 sec')
```

Comment:
The third parameter to these calls is identical to the first - suggest the third one is dropped.  The message on failure will still be readable.